packages/ide-extension/src/telemetry/telemetry.ts

```diff
@@ -0,0 +1,148 @@
+import { platform, arch, release } from 'os';
+import { env, workspace } from 'vscode';
+import type { Disposable } from 'vscode';
+import { TelemetryClient } from 'applicationinsights';
+import type { Contracts } from 'applicationinsights';
+import { logString } from '../logger/logger';
+import packageJson from '../../package.json';
+import type { TelemetryEvent, TelemetryReporter } from '../utils/telemetry';
+
+const key = 'ApplicationInsightsInstrumentationKeyPLACEH0LDER';
+
+// Telemetry reporter client
+let reporter: TelemetryReporter | undefined;
+
+/**
+ * Initialize telemetry.
+ *
+ * @returns - telemetry reporter
+ */
+export function initTelemetry(): TelemetryReporter {
+    const disposables: Disposable[] = [];
+    if (!reporter) {
+        const client = new TelemetryClient(key);
+        client.channel.setUseDiskRetryCaching(true);
+        client.addTelemetryProcessor((envelope: Contracts.Envelope) => {
+            envelope.tags['ai.location.ip'] = '0.0.0.0';
+            envelope.tags['ai.cloud.roleInstance'] = 'masked';
+            return true;
+        });
```

Comment:
Please add a comment regarding the tags used here. It isn't clear what these tags mean or why they are required.